src/renderer/shared/pallet/proxy/storage.ts

```diff
@@ -0,0 +1,82 @@
+import { type ApiPromise } from '@polkadot/api';
+import { zipWith } from 'lodash';
+
+import { substrateRpcPool } from '@/shared/api/substrate-helpers';
+import { type AccountId, pjsSchema } from '@/shared/polkadotjs-schemas';
+
+import { proxyAnnouncement, proxyProxyDefinition } from './schema';
+
+const getQuery = (api: ApiPromise, name: string) => {
+  const pallet = api.query['proxy'];
+  if (!pallet) {
+    throw new TypeError(`proxy pallet not found in ${api.runtimeChain.toString()} chain`);
+  }
+
+  const query = pallet[name];
+
+  if (!query) {
+    throw new TypeError(`${name} query not found`);
+  }
+
+  return query;
+};
+
+export const storage = {
+  /**
+   * The announcements made by the proxy (key).
+   */
+  announcements(api: ApiPromise, accounts?: AccountId[]) {
+    const recordSchema = pjsSchema.tupleMap(
+      ['announcements', pjsSchema.vec(proxyAnnouncement)],
+      ['deposit', pjsSchema.u128],
+    );
+
+    if (accounts && accounts.length > 0) {
+      const schema = pjsSchema.vec(recordSchema);
+
+      return substrateRpcPool
+        .call(() => getQuery(api, 'announcements').multi(accounts))
+        .then(schema.parse)
+        .then(result => zipWith(accounts, result, (account, value) => ({ account, value })));
+    } else {
```

Comment:
Is this else section really needed? we have return in if section. Same for others